### ISSUE TITLE
Infinite processing for some items

### ISSUE THREAD

**Issue by Winiciusf** [sepinf-inc/IPED#702]
Hello!

I'm analyzing a MacBook. First, I created one DMG, using the Apple Disk Utility. After, I added this image file in FTK and I created a AD1 image, with some folders. Then, I started processing this image with IPED. However, after some hours, I noticed that the processing had stopped on some files. The status bar is stopped, but the software seems to be working.

Please, any idea for solving this problem? 

Thank you!

[IPED-2021-08-16-20-05-47.log](https://github.com/sepinf-inc/IPED/files/6995875/IPED-2021-08-16-20-05-47.log)
[IPED-2021-08-16-20-18-18.log](https://github.com/sepinf-inc/IPED/files/6995876/IPED-2021-08-16-20-18-18.log)

![Status](https://user-images.githubusercontent.com/68752755/129641398-64d1674d-1efa-4e68-9982-2187c0e1631e.PNG)

**Comment by lfcnassif** [comment]
For this kind of issue we need a thread dump as described here: https://github.com/sepinf-inc/IPED/wiki/Troubleshooting#processing-frozen
Could you take it?

And I see some "access denied" errors in your logs, maybe this could be related to antivirus software blocking things, the thread dump maybe could help...

**Comment by lfcnassif** [comment]
And I don't see "processing hours" in your logs, they show less than 2 minutes of processing time.

**Comment by Winiciusf** [comment]
Really I used the option "--continue". This is the fist log file
[IPED-2021-08-15-13-03-42.log](https://github.com/sepinf-inc/IPED/files/6995973/IPED-2021-08-15-13-03-42.log)

**Comment by Winiciusf** [comment]
Really I used the "--continue" option. This first log is this:
[IPED-2021-08-16-08-31-44.log](https://github.com/sepinf-inc/IPED/files/6995988/IPED-2021-08-16-08-31-44.log)

**Comment by lfcnassif** [comment]
Still the wrong log. Anyway, we need the thread dump.

**Comment by Winiciusf** [comment]
[java_.zip](https://github.com/sepinf-inc/IPED/files/6996083/java_.zip)

**Comment by lfcnassif** [comment]
This isn't useful, please take the thread dump using:
jstack -l <pid>

**Comment by Winiciusf** [comment]
Sorry!
[java.zip](https://github.com/sepinf-inc/IPED/files/6996260/java.zip)
Is it correct?

**Comment by Winiciusf** [comment]
Other doubt: Is it possible to stop/finish the processing but without alerts when opening the search interface?

**Comment by Winiciusf** [comment]
Other question: I tried to use the https://github.com/sepinf-inc/IPED/wiki/Troubleshooting#processing-frozen. I updated the file "\profiles\en\default\conf\TaskInstaller.xml" and I created the file "\profiles\en\default\conf\scripts\IgnoreFile.js". However, the task "IgnoreFile" was not loaded. Any suggestion, Nassif?

[conf.zip](https://github.com/sepinf-inc/IPED/files/6999401/conf.zip)

![image](https://user-images.githubusercontent.com/68752755/129715313-637e463e-00dd-4874-a933-12f075a61048.png)

**Comment by lfcnassif** [comment]
> Is it correct?

Yes, thanks. Analysing them, I don't see any aparent problem in these thread dumps, relevant threads seems to be running. Aren't the number of processed items being increased in the processing window and items aren't changing? If not, I think this could be related to many (Acesso negado) messages in your logs, maybe this is some kind of permission problem. Check if there is some process blocking those files (with "acesso negado" messages), maybe antivirus or old iped processes left behind...

> Is it possible to stop/finish the processing but without generate alerts in the search interface?

If you do this, you will end up with missing items, including allocated ones, so no, the app always warn the user about this situation.

> Any suggestion, Nassif?

You should do this inside the "indexador" folder in ouput case folder. When you are using --continue or --append, the existing case configurations are used.

**Comment by lfcnassif** [comment]
> Aren't the number of processed items being increased in the processing window and items aren't changing?

To clarify, the progress bar is just updated when containers are fully processed. So if a zip or database is being expanded (extracting subitems and blobs), the total/processed item count is increased and current items being processed change, but the progress bar is just updated when the full zip/database finishes. I think this could be happing and maybe you just have to wait some more time, that can go from minutes to hours depeding on your hardware.

**Comment by Winiciusf** [comment]
Nassif, thank you!

In other moment this processing had 91 hours. Then I think strange, because the "Current Items" was processing the same files for many hours. In this case, the status bar was stuck, but the "Exported Items" really was slowly increasing. 
Anyway your orientation about the correct local of the script file .js was very important!
I will process again with patience. Finally, if necessary, I will use the "IgnoreFile.js" and "TaskInstaller.xml" (now in the correct folder).

On weekend I will post the final result.

Nassif, thank you so much! Congratulations for the job!

**Comment by lfcnassif** [comment]
> In other moment this processing had 91 hours

A thread dump at this moment could be helpful.

> but the "Exported Items" really was slowly increasing

So, the processing wasn't stuck, just slow.

> On weekend I will post the final result.

Ok

**Comment by Winiciusf** [comment]
Nassif, I excluded about 10 files from processing using the IgnoreFile.js. Then, the processing was finished sucessful. 
Thank you!!!

**Comment by lfcnassif** [comment]
Well, ideally we should still investigate the cause of this. If you can send me the files privately, I can take a look. Or if you can start the processing again from scratch without excluding them during the weekend and take a thread thump on next Monday, that maybe can help.

**Comment by Winiciusf** [comment]
Off course!
Nassif, I will send to you those files. Could I send it to your email @dpf or lfcnassif@gmail.com?

**Comment by lfcnassif** [comment]
Great! Please send a link to them to @pf.gov.br

**Comment by lfcnassif** [comment]
Just an update, @Winiciusf sent me the files. I was not able to reproduce, processing finished in ~2min with the suspect files. He will try to reproduce in another machine to check if it is an environment issue.
